x-pack/plugins/infra/server/routes/custom_dashboards/get_custom_dashboard.ts

```diff
@@ -0,0 +1,54 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0; you may not use this file except in compliance with the Elastic License
+ * 2.0.
+ */
+
+import { createRouteValidationFunction } from '@kbn/io-ts-utils';
+import {
+  InfraGetCustomDashboardsRequestParamsRT,
+  InfraGetCustomDashboardsResponseBodyRT,
+} from '../../../common/http_api/custom_dashboards_api';
+import { KibanaFramework } from '../../lib/adapters/framework/kibana_framework_adapter';
+import { handleRouteErrors } from '../../utils/handle_route_errors';
+import { checkCustomDashboardsEnabled } from './lib/check_custom_dashboards_enabled';
+import { findCustomDashboard } from './lib/find_custom_dashboard';
+
+export function initGetCustomDashboardRoute(framework: KibanaFramework) {
+  const validateParams = createRouteValidationFunction(InfraGetCustomDashboardsRequestParamsRT);
+
+  framework.registerRoute(
+    {
+      method: 'get',
+      path: '/api/infra/custom-dashboards/{assetType}',
+      validate: {
+        params: validateParams,
+      },
+    },
+    handleRouteErrors(async (context, request, response) => {
```

Comment:
I think the name `handleRouteErrors` is a bit odd. At first, I thought this would be handled only when errors happened. But that is not the case, right? It wraps up the handle and handles when errors happen.

Reply:
Does it make sense to implement it [here](https://github.com/elastic/kibana/blob/main/x-pack/plugins/infra/server/lib/adapters/framework/kibana_framework_adapter.ts#L52-L53) instead?

Reply:
> I think the name handleRouteErrors is a bit odd. At first, I thought this would be handled only when errors happened. But that is not the case, right? It wraps up the handle and handles when errors happen.

Could you explain a bit more? Not sure I get your point.

> Does it make sense to implement it [here](https://github.com/elastic/kibana/blob/main/x-pack/plugins/infra/server/lib/adapters/framework/kibana_framework_adapter.ts#L52-L53) instead?

Yeah, I mentioned this in the PR description. It would definitely make sense but just a bit out of scope for this change I think.
> In the future we could do something similar right within `registerRoutes` framework function, but this would require a bit of refactoring.

Reply:
>Added a wrapper handleRouteErrors to avoid error handling duplication which we now have in a few routes. In the future we could do something similar right within registerRoutes framework function, but this would require a bit of refactoring.

Can you elaborate on why this would need refactoring?

Reply:
Mainly because some routes already have a bespoke error handling and ideally we'd need to remove those and re-test the routes.

Reply:
If you can create a follow-up issue for this tech debt would be nice, so we don't lose track of it.

Reply:
Created [the issue](https://github.com/elastic/kibana/issues/176993)